Code/Robot/Robot/Robot.ino

```diff
@@ -135,136 +217,219 @@ void read_serial(JsonDocument& doc) {
   }
 }
 
-void driving_logic(std::queue<Move>* moveQueue) {
-  float wheelSpeeds[cNumberOfWheels];  // Initialize motor speeds
+void executeMoveSequence(std::queue<Move>* moveQueue) {
+  while (!moveQueue->empty()) {  // Loop until the queue is empty
+    Move nextMove = getNextMoveFromQueue(moveQueue);
+    switch ((MoveType)nextMove.moveType) {
+      case eFreeDrive:
+        executeFreeDrive(nextMove);
+        break;
+      case eLineFollow:
+        executeLineFollow(nextMove);
+        break;
+      case eScissor:
+        executeScissor(nextMove);
+        break;
+      case eBelt:
+        executeBelt(nextMove);
+        break;
+      case eCalibrate:
+        calibrate(nextMove);
+        break;
+      default:
+        DEBUG_PRINT("Unexpected moveType: ");
+        DEBUG_PRINTLN(nextMove.moveType);
+        break;
+    }
```

Comment:
```suggestion
    switch (nextMove.moveType) {
      case eFreeDrive:
        executeFreeDrive(nextMove);
        break;
      case eLineFollow:
        executeLineFollow(nextMove);
        break;
      case eScissor:
        executeScissor(nextMove);
        break;
      case eBelt:
        executeBelt(nextMove);
        break;
      case eCalibrate:
        calibrate(nextMove);
        break;
      default:
        DEBUG_PRINT("Unexpected moveType: ");
        DEBUG_PRINTLN(nextMove.moveType);
        break;
    }
```

cast no longer necessary because we're storing it as our enum in the struct